client/components/domains/register-domain-step/index.jsx

```diff
@@ -584,20 +583,36 @@ const RegisterDomainStep = React.createClass( {
 				}
 				break;
 
-			case 'mappable_but_forbidden_subdomain':
+			case 'not_mappable_forbidden_subdomain':
 				message = this.translate( 'Subdomains starting with \'www.\' cannot be mapped to a WordPress.com blog' );
 				break;
 
-			case 'mappable_but_mapped_domain':
+			case 'not_mappable_invalid_tld':
+			case 'invalid_query':
+				message = this.translate( 'Sorry, %(domain)s does not appear to be a valid domain name.', {
+					args: { domain: domain }
+				} );
+				break;
+
+			case 'not_mappable_mapped_domain':
 				message = this.translate( 'This domain is already mapped to a WordPress.com site.' );
 				break;
 
-			case 'mappable_but_restricted_domain':
+			case 'not_mappable_restricted_domain':
 				message = this.translate(
 					'You cannot map another WordPress.com subdomain - try creating a new site or one of the custom domains below.'
 				);
 				break;
 
+			case 'not_mappable_recently_mapped':
+				message = this.translate( 'This domain has been recently mapped by a different user - please try again later.' );
```

Comment:
Sorry, should have said `unmapped` (I got tricked by the unmappability reason coming from the backend). Context: we don't allow mapping a domain again just after it has been unmapped by a _different_ user. The previous message just read `This domain cannot currently be mapped.`, so I wanted something more explicit.